Added integration with eglot

RuijieYu opened the pull request:
Fixes #121.

* gdscript-eglot.el gdscript-eglot-version: added customizable option for the godot version in use.
(gdscript-eglot-contact): added function to allow `eglot` to find the godot LSP port.

RuijieYu commented:
I don't have the proper environment for this, @xiliuya (or someone else), can you test this PR and see if it works? 

I'm unsure about the header.  It seems that there is an "author" field, but I don't know if that is reserved for the entire repository or for authoring each file. 

This PR also took inspiration from code by @xiliuya, and I'm wondering whether/how I should properly attribute that.

joaotavora commented:
I think you should also add something to `gdscript-mode.el` that `require`'s this file, checks if the `eglot-server-programs` variable exists and uses `add-to-list` to add the new function to that variable.

RuijieYu commented:
> I think you should also add something to `gdscript-mode.el` that `require`'s this file, checks if the `eglot-server-programs` variable exists and uses `add-to-list` to add the new function to that variable.

I added the require expr in gdscript-mode.  Also added a push that adds gdscript to `eglot-server-programs` as soon as `eglot` is loaded.  I think that's enough?  I didn't put a lambda wrapping `gdscript-eglot-contact` because that function is autoloaded from the user session, and also `require`d from within the `-mode` file.

NathanLovato commented:
Thanks much for taking the time to make this contribution.

Note that I I'm not maintaining this module anymore, I don't use Emacs anymore, so this project needs a new maintainer who actively uses Emacs to review, test, and merge your changes.

xiliuya commented:
> I don't have the proper environment for this, @xiliuya (or someone else), can you test this PR and see if it works?
> 
> I'm unsure about the header. It seems that there is an "author" field, but I don't know if that is reserved for the entire repository or for authoring each file.
> 
> This PR also took inspiration from code by @xiliuya, and I'm wondering whether/how I should properly attribute that.

Great, I don't need to change the configuration file.

RuijieYu commented:
@NathanLovato Given the situation, do you want to make a pinned issue on this repository to look for someone who wants to take over the maintainership?  I have never used gdscript nor this code, so I definitely wouldn't qualify.  Unless @xiliuya wants to take over the maintainership for this repo, of course.

xiliuya commented:
@RuijieYu Yes, I want to become the maintainer of this repository.
Now I have replied in relevant [issues/123](https://github.com/godotengine/emacs-gdscript-mode/issues/123)

xiliuya commented:
Hello, @RuijieYu , Thank you very much for your submission.
I found the following problems in my test, which need your modification:
- eglot-server-programs needs the right pattern:
```diff
diff --git a/gdscript-mode.el b/gdscript-mode.el
index b6c5556..6f626c6 100644
--- a/gdscript-mode.el
+++ b/gdscript-mode.el
@@ -53,7 +53,7 @@
 ;;;###autoload
 (with-eval-after-load 'eglot
   (defvar eglot-server-programs)
-  (push (cons 'eglot #'gdscript-eglot-contact)
+  (push (cons 'gdscript-mode #'gdscript-eglot-contact)
         eglot-server-programs))
 
 (defvar gdscript-mode-map (let ((map (make-sparse-keymap)))

```

xiliuya commented:
@RuijieYu  
`gdscript-eglot.el` need commentary.
As described below: 
```
;;; Commentary:
;;
;; Handles the configuration of eglot.
;; It supports gdscript-mode using eglot.
;;
```

RuijieYu commented:
Fixed.  Wasn't sure what I was thinking back then about `push eglot`...

xiliuya commented:
@NathanLovato This PR is ready to merge into master.

RuijieYu commented:
Almost forgot that I had that TODO there.  Also, fixed a docstring inconsistency (used to say `INTERACTIVE-P` but should be `INTERACTIVE` instead).  Now I think it is ready.

RuijieYu commented:
Seems like something is unhappy about eask on Windows.  Is it due to the PR, or something else?

xiliuya commented:
@RuijieYu I think it's will fix with [#118](https://github.com/godotengine/emacs-gdscript-mode/pull/118)

xiliuya reviewed (commented):
This supports `gdscript-mode' using `eglot'.

xiliuya reviewed (approved):
This supports `gdscript-mode' using `eglot'.

xiliuya reviewed (approved):
This supports `gdscript-mode' using `eglot'.

NathanLovato commented:
Thank you for the contribution! The windows-latest test doesn't pass currently but it's just incompatible with eask it seems. As mentioned in #118 a solution is to add an experimental flag to bypass this (as it is for an experimental Emacs version, hence the incompatibility).

So as review passed here, I'm merging. Thank you very much for your work, you two!
